teapots/choreographer-30fps/src/main/cpp/CMakeLists.txt

```diff
@@ -17,36 +17,53 @@
 cmake_minimum_required(VERSION 3.4.1)
 
 # build native_app_glue as a static lib
-add_library(native_app_glue STATIC
-    ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
-
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
+add_library(NativeAppGlue STATIC
+  ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
 
 # build the ndk-helper library
-set(ndk_helper_dir ../../../../common/ndk_helper)
-add_subdirectory(${ndk_helper_dir} ndk_helper)
-
-# Export ANativeActivity_onCreate(), 
-# Refer to: https://github.com/android-ndk/ndk/issues/381.
-set(CMAKE_SHARED_LINKER_FLAGS
-    "${CMAKE_SHARED_LINKER_FLAGS} -u ANativeActivity_onCreate")
+set(ndkHelperSrc ../../../../common/ndk_helper)
+add_subdirectory(${ndkHelperSrc} ndkHelperBin)
 
 # now build app's shared lib
-add_library(ChoreographerNativeActivity SHARED
+add_library(ChoreographerNativeActivity
+  SHARED
     ChoreographerNativeActivity.cpp
-    TeapotRenderer.cpp)
+    TeapotRenderer.cpp
+)
```

Comment:
I like trying to separate out common code from platform specific code.  Might I suggest that TeapotRenderer.cpp can be common to any iOS/Android style targeting at some point in the future; thus structure the lib to have something like this:
add_library(ChoreographerNativeActivity SHARED TeapotRenderer.cpp)
if(ANDROID)
add_library(androidActivityLib STATIC ChoreographerNativeActivity.cpp)
target_link_libraries(ChoreographerNativeActivity ActivityLib)
endif()

Reply:
I think given that the goal of this sample is to show how to use the choreographer API (unless I'm assuming too much based on the directory name), it's best to keep it focused on that. If TeapotRenderer.cpp were generalized and shared among multiple samples that would be a different story, but I think remaining concise is better here.

Reply:
Admittedly it's probably better to just make an independent crops platform example